libs/ts/firebase/adapter/src/lib/utilities/firebase-service-factory.utility.ts

```diff
@@ -0,0 +1,26 @@
+import { FirebaseSdkMethod } from '../models/firebase-sdk-method.type';
+import { FirebaseServiceMethod } from '../models/firebase-service-method.type';
+
+type FirebaseServiceMethods<T> = {
+    [key: string]: FirebaseSdkMethod<T, any, any>;
+};
+
+export class FirebaseServiceFactoryUtility {
+    static create<T, Methods extends FirebaseServiceMethods<T>>(
```

Comment:
This factory adapts the pure Firebase SDK methods to be suitable for Angular by providing the "instance" dependency relevant for the methods and bundling them into an injectable object.